### PR TITLE
fix outdated sample resources, add crd installation to readme

### LINKED ISSUE #12: Missing steps in readme, outdated sample custom resources
I think there are a couple of issues in the readme that need to be fixed. 

The current first steps recommends to install the custom resource samples. This will of course not work because there was no step installing the CRDS. Error:

```
error validating "config/samples/kustomization.yaml": error validating data: [apiVersion not set, kind not set]; if you choose to ignore these errors, turn validation off with --validate=false
Error from server (NotFound): error when creating "config/samples/perses.dev_v1alpha1_perses.yaml": the server could not find the requested resource (post perses.perses.dev)
Error from server (NotFound): error when creating "config/samples/perses.dev_v1alpha1_persesdashboard-2.yaml": the server could not find the requested resource (post persesdashboards.perses.dev)
Error from server (NotFound): error when creating "config/samples/perses.dev_v1alpha1_persesdashboard.yaml": the server could not find the requested resource (post persesdashboards.perses.dev)
Error from server (NotFound): error when creating "config/samples/perses.dev_v1alpha1_persesdatasource.yaml": the server could not find the requested resource (post persesdatasources.perses.dev)
```

I think there should be step `make install` before this.

Even with CRDs installed, the step `kubectl apply -f config/samples/` does not work:

```
error validating "config/samples/kustomization.yaml": error validating data: [apiVersion not set, kind not set]; if you choose to ignore these errors, turn validation off with --validate=false
Error from server (Invalid): error when creating "config/samples/perses.dev_v1alpha1_perses.yaml": Perses.perses.dev "perses-sample" is invalid: spec.config.deactivate_front: Required value
Error from server (BadRequest): error when creating "config/samples/perses.dev_v1alpha1_persesdashboard-2.yaml": PersesDashboard in version "v1alpha1" cannot be handled as a PersesDashboard: strict decoding error: unknown field "spec.kind", unknown field "spec.metadata", unknown field "spec.spec"
Error from server (BadRequest): error when creating "config/samples/perses.dev_v1alpha1_persesdashboard.yaml": PersesDashboard in version "v1alpha1" cannot be handled as a PersesDashboard: strict decoding error: unknown field "spec.kind", unknown field "spec.metadata", unknown field "spec.spec"
Error from server (BadRequest): error when creating "config/samples/perses.dev_v1alpha1_persesdatasource.yaml": PersesDatasource in version "v1alpha1" cannot be handled as a PersesDatasource: strict decoding error: unknown field "spec.kind", unknown field "spec.metadata", unknown field "spec.spec"
```

I think the sample custom resources have not been updated in a while and are now outdated.

### PR DESCRIPTION
**Pull request by basti1302**
fixes #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,34 +9,27 @@ You’ll need a Kubernetes cluster to run against. You can use [KIND](https://si
 
 ### Running on the cluster
 
-1. Install Instances of Custom Resources:
-
+1. Install custom resource definitions:
 ```sh
-kubectl apply -f config/samples/
+make install
```
Comment (basti1302) [inline README.md:14]: added this as the new first step

File: README.md [hunk 2]
```diff
@@ -9,34 +9,27 @@ You’ll need a Kubernetes cluster to run against. You can use [KIND](https://si
...
 ```
 
-2. Build and push your image to the location specified by `IMG`:
+2. Install custom resources:
 
 ```sh
-make docker-build docker-push IMG=<some-registry>/perses-operator:tag
+bin/kustomize build config/samples | kubectl apply -f -
```
Comment (basti1302) [inline README.md:20]: I think directly applying the config/samples dir via kubectl is not correct, the resources are supposed to be rendered by kustomize, right?

File: README.md [hunk 3]
```diff
@@ -9,34 +9,27 @@ You’ll need a Kubernetes cluster to run against. You can use [KIND](https://si
...
 ```
 
-3. Deploy the controller to the cluster with the image specified by `IMG`:
+3. Build and push your image to the location specified by `IMG`:
```
Comment (basti1302) [inline README.md:23]: this is the previous step 2 (unchanged)

File: README.md [hunk 4]
```diff
@@ -9,34 +9,27 @@ You’ll need a Kubernetes cluster to run against. You can use [KIND](https://si
...
 ```sh
-make deploy IMG=<some-registry>/perses-operator:tag
-```
-
-4. Install a CRD instance
-
-```sh
-kubectl apply -f config/samples/v1alpha1_perses.yaml --namespace default
+make docker-build docker-push IMG=<some-registry>/perses-operator:tag
 ```
```
Comment (basti1302) [inline README.md:27]: I removed this step. It is not clear to me why you would want to do this. Applying the whole directory `config/samples` (which is mentioned earlier) already does this?

File: README.md [hunk 5]
```diff
@@ -9,34 +9,27 @@ You’ll need a Kubernetes cluster to run against. You can use [KIND](https://si
...
 
-5. Install a CRD instance
+4. Deploy the controller to the cluster with the image specified by `IMG`:
 
 ```sh
-kubectl delete -f config/samples/v1alpha1_perses.yaml --namespace default
```
Comment (basti1302) [inline README.md:39]: I removed this as well -- it was labelled "install" but actually removes one specific resource?

File: config/samples/kustomization.yaml [hunk 6]
```diff
@@ -1,4 +1,6 @@
 ## Append samples you want in your CSV to this file as resources ##
 resources:
-  - v1alpha1_perses.yaml
+  - perses.dev_v1alpha1_perses.yaml
+  - perses.dev_v1alpha1_persesdashboard.yaml
+  - perses.dev_v1alpha1_persesdatasource.yaml
```
Comment (basti1302) [inline config/samples/kustomization.yaml:5]: I added one resource for each CRD here.

File: config/samples/perses.dev_v1alpha1_perses.yaml [hunk 7]
```diff
@@ -19,4 +19,5 @@ spec:
       queries_path: "/etc/perses/schemas/queries"
       datasources_path: "/etc/perses/schemas/datasources"
       variables_path: "/etc/perses/schemas/variables"
+    deactivate_front: false
```
Comment (basti1302) [inline config/samples/perses.dev_v1alpha1_perses.yaml:22]: required property

File: config/samples/perses.dev_v1alpha1_persesdashboard-2.yaml [hunk 8]
```diff
@@ -9,555 +9,547 @@ metadata:
     app.kubernetes.io/created-by: perses-operator
   name: perses-dashboard-sample-2
 spec:
-  kind: Dashboard
-  metadata:
-    name: Demo-2
-    project: default
-    createdAt: "2021-11-09T00:00:00Z"
-    updatedAt: "2021-11-09T00:00:00Z"
-    version: 0
```
Comment (basti1302) [inline config/samples/perses.dev_v1alpha1_persesdashboard-2.yaml:18]: I think this has not been updated after https://github.com/perses/perses-operator/commit/3a8e2193c0b03f2877286a01e64b391e5610cb2b. Now the resource is valid again. Same for the other dashboard resource.

File: config/samples/perses.dev_v1alpha1_persesdatasource.yaml [hunk 9]
```diff
@@ -9,18 +9,10 @@ metadata:
...
   name: perses-datasource-sample
 spec:
-  kind: Datasource
-  metadata:
-    name: perses-prometheus-datasource
-    project: default
-    createdAt: "2021-11-09T00:00:00Z"
-    updatedAt: "2021-11-09T00:00:00Z"
-    version: 0
-  spec:
```
Comment (basti1302) [inline config/samples/perses.dev_v1alpha1_persesdatasource.yaml:19]: This resource was also invalid and is now valid.

**Comment by basti1302** [comment]
Thanks for the update with regard to the state of this repository. Unfortunately I do not have the capacity to contribute significantly to this project.

The CI build issues are unrelated to this PR.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa[40](https://github.com/perses/perses-operator/actions/runs/11124716530/job/30958820291#step:5:41)aef]
```

IIRC this is a known bug either in the operator framework or some related dependency. Could be this one, not sure: https://github.com/operator-framework/operator-sdk/issues/6230

Locally I bumped CONTROLLER_TOOLS_VERSION` from `v0.11.1` to `v0.14.0` to get the build working.

I can create a separate PR, however, that will also regenerate things in `config`, which is why I left these changes out of this PR.

**Comment by Nexucis** [comment]
no worries it was just in case `Dash0` wanted to see it released a bit sooner.

Thanks for checking @basti1302

**Comment by basti1302** [comment]
I created a PR for updating the controller tools version here: https://github.com/perses/perses-operator/pull/14

File: README.md (hunk 2 above)
Comment (jgbernalp) [reply README.md:20]: yes, correct.

**Review by jgbernalp — APPROVED** [review]

**Comment by jgbernalp** [comment]
LGTM, we can merge when the pipeline is fixed with #14

**Comment by Nexucis** [comment]
PR #14 has been merged. @basti1302 Can you rebase your PR to see if the build is passing ?

Thanks for your work here, that is super appreciated !

**Comment by basti1302** [comment]
> PR #14 has been merged. @basti1302 Can you rebase your PR to see if the build is passing ?
> 
> Thanks for your work here, that is super appreciated !

Rebased, you're welcome 👍

**Review by Nexucis — APPROVED** [review]
awesome !

**Pull request merged**